### PR TITLE
Purchase url

### PR DESCRIPTION
**Pull request by Jaapp-**
Add purchaseUrl and purchaseTitle to SoundCloudTrack, database, and API.
Frontend PR: https://github.com/MoodCat/MoodCat.me-Frontend/pull/53

### CODE REVIEW

File: src/main/java/me/moodcat/soundcloud/SoundCloudExtract.java [hunk 1]
```diff
@@ -13,6 +13,12 @@
  */
 public class SoundCloudExtract extends SoundCloudAPIConnector {
 
+    public static void main(String[] args) throws SoundCloudException {
```
Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/soundcloud/SoundCloudExtract.java]: I suppose that this was a dev test? Maybe it should be removed or moved to a (integration) test case

**Comment by JWGmeligMeyling** [comment]
One remark, but the PR looks good!

File: src/main/java/me/moodcat/soundcloud/SoundCloudExtract.java (hunk 1 above)
Comment (Jaapp-) [inline src/main/java/me/moodcat/soundcloud/SoundCloudExtract.java]: Ah, yes, that should not be there. I'll remove it.

File: src/main/java/me/moodcat/api/models/SongModel.java [hunk 2]
```diff
@@ -73,6 +73,24 @@
     private double arousal;
 
     /**
+     * The link to the site to purchase the song.
+     *
+     * @param purchaseUrl
+     *          The new purchaseURL to set.
```
Comment (Azkle) [inline src/main/java/me/moodcat/api/models/SongModel.java]: I would delete the new in the javadoc (wich is almost all javadocs)
This insist that this isn't used for a initial purchaseUrl.

**Comment by Azkle** [comment]
LGTM, just one very minor jdoc thing.

File: src/main/java/me/moodcat/api/models/SongModel.java [hunk 3]
```diff
@@ -73,6 +73,24 @@
...
+     * @return The url to the artwork image.
```
Comment (GijsWeterings) [inline src/main/java/me/moodcat/api/models/SongModel.java]: url of the artwork image?

File: src/main/java/me/moodcat/api/models/SongModel.java (hunk 2 above)
Comment (Jaapp-) [inline src/main/java/me/moodcat/api/models/SongModel.java]: I'm not sure about this one, but I think it is out of scope of this PR as it affects a lot of Javadoc that doesn't have to do with the purchase URL.

File: src/main/java/me/moodcat/api/models/SongModel.java (hunk 3 above)
Comment (Jaapp-) [inline src/main/java/me/moodcat/api/models/SongModel.java]: Whoops

File: src/main/java/me/moodcat/api/models/SongModel.java (hunk 2 above)
Comment (Azkle) [inline src/main/java/me/moodcat/api/models/SongModel.java]: I only ment the javadoc from this commit.

**Comment by JeremybellEU** [comment]
This PR should be rebased to master in order to be mergable.

**Pull request closed without merging**